### ISSUE TITLE
Rules to enforce either doc comments or docstrings for variables/attributes

### ISSUE THREAD

**Issue by sh-at-cs** [astral-sh/ruff#28453]
#### Background

There are two distinct conventions in Python for attaching documentation to module-level variables and class attributes:

1. A docstring directly below, e.g.:
    ```python
    name: str
    """
    The user's full name.

    Should not include middle names.
    """
    ```

2. A `#:`[^1] [doc comment](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#doc-comments-and-docstrings) directly above, e.g.:
    ```python
    #: The user's full name
    #:
    #: Should not include middle names.
    name: str
    ```

I believe the docstring-below approach (1) is more common nowadays.

The doc-comment-above approach (2) was (according to an LLM's recollections) originally introduced by [epydoc](https://epydoc.sourceforge.net/epydoc.html#variable-docstrings) and then adopted by Sphinx.

#### Rule suggestion

It would be nice if there were rules to enforce either one of these approaches, complaining if the other one is used for a variable/attribute.

If you don't want to implement a rule to enforce doc comments because they seem kind of outdated anyway, it would also be fine to only have a rule that enforces docstrings and complains about doc comments.

[^1]: Spelled out so people may find this issue more easily: *hash colon*
